utils/litemap/src/map.rs

```diff
@@ -144,6 +146,9 @@ impl<K: Ord, V> LiteMap<K, V> {
             if last.0 > key {
                 return Some((key, value));
             }
+            if last.0 == key {
+                self.values.pop();
```

Comment:
This is a good catch.  However, it seems messy if we silently throw out an entry from the map.  How about making the first condition be `if last.0 >= key` and then just failing if the key is the same as a preexisting key.

Reply:
I did this to make `extend_from_sorted()` and `extend_from_unsorted()` behave the same if the list is all the same. 

Let's say we have an iterator that is (pseudo-formatting):
```
Iterator: (1, "one") -> (1, "two") -> (1, "three")
```

If we called `extend_from_unsorted()` on this iterator (where all the keys are the same) it just calls `insert()` every time, which will overwrite the key, and the final value in the map will be `"three"`. 

But if `try_append()` _doesn't_ overwrite equal key values, and we called `try_extend_from_sorted()`  it would stop at the first duplicate, leaving `"one"` in the map, and the other two items in the remaining iterator. And that might be fine. 

Thoughts?

Reply:
That's a good question. `insert()` overwrites the key, but it returns the value back out again. I think `try_append()` should be more strict than that, and just refuse to append something that isn't strictly increasing.  Which would mean that `try_extend_from_sorted()` should also be strict, and `extend_from_unsorted()` can be "lenient", like `insert()`.

Reply:
I agree with that. I've updated it to fail on equal keys.